File: practice31/6.cpp


//https://www.nowcoder.com/practice/8e0fdf50f69f42458cb0aa043d9c41d6?tpId=90&tqId=30868&tPage=5&rp=5&ru=/ta/2018test&qru=/ta/2018test/question-ranking
#include<iostream>
#include<string>
#include<algorithm>
using namespace std;

int main() {
	int n;
	cin >> n;
	int *pt = new int[n];
	for (int i = 0; i < n; i++) {
		cin >> pt[i];
	}
	int index, temp, x, sign;
	if (pt[0] == 1) {
		sign = 0;
		for (int i = 1; i < n; i++) {
			if (pt[i] == pt[i - 1] + 1) {
				continue;
			}
			else {
				x = i;
				sign = 1;
				break;
			}
		}
		for (int i = x + 1; i < n; i++) {
			if (pt[i] == pt[x - 1] + 1) {
				index = i;
				break;
			}
		}
		if (sign == 1) {
			temp = pt[x], pt[x] = pt[index], pt[index] = temp;
		}
		else {
			temp = pt[n - 2], pt[n - 2] = pt[n - 1], pt[n - 1] = temp;
		}
	}
	else {
		for (int i = 0; i < n; i++) {
			if (pt[i] == 1) {
				index = i;
				break;
			}
		}
		temp = pt[0], pt[0] = pt[index], pt[index] = temp;
	}
	for (int i = 0; i < n - 1; i++) {
		cout << pt[i] << " ";
	}
	cout << pt[n - 1] << endl;
	system("pause");
	return 0;
}

File: practice33/6.java


//https://www.nowcoder.com/practice/bd891093881d4ddf9e56e7cc8416562d?tpId=85&tqId=29864&tPage=2&rp=2&ru=/ta/2017test&qru=/ta/2017test/question-ranking
import java.math.BigInteger;
import java.util.*;
import java.util.Scanner;

public class Main {
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        String str = sc.nextLine();
        int index;
        String temp, result="";
        for (int i = 0; i < str.length(); i++) {
            if (str.charAt(i) >= '0' && str.charAt(i) <= '9') {
                index = i;
                while (i < str.length() && str.charAt(i) >= '0' && str.charAt(i) <= '9') {
                    i++;
                }
                temp = str.substring(index, i);
                if(temp.length()>result.length()){
                    result=temp;
                }
            }
        }
        System.out.println(result);
    }
}